.github/workflows/tests.yml

```diff
@@ -1,27 +1,117 @@
-name: Tests
+name: CI Workflow for tests
 
 on:
+  pull_request_target:
+    types: [opened, synchronize, reopened, labeled]
   push:
-    branches:
-      - master
-  pull_request:
     branches: [ master ]
 
-concurrency:
-  group: ci-${{ github.ref }}-test
-  cancel-in-progress: true
```

Comment:
Why remove this early cancellation of already running tests when a new commit is pushed in quick succession on a PR?

Reply:
Nice spot, it was an unwanted removal indeed

Reply:
We're entering Nirvana territory :)